packages/controller/src/Controller.ts

```diff
@@ -165,6 +166,31 @@ export default class Controller {
 		this.app.locals.controller = this;
 		this.app.locals.streams = new Map();
 
+		// Log all requests made to express
+		this.app.use((req, res, next) => {
+			const startNs = process.hrtime.bigint();
+			res.once("finish", () => {
+				const endNs = process.hrtime.bigint();
+				const durationMs = (Number(endNs - startNs) / 1e6);
+				logger.log({
+					level: "http",
+					message: `HTTP${req.httpVersion} ${req.method} ${req.originalUrl}` +
+						` (Status ${res.statusCode} in ${durationMs}ms)`,
+					meta: {
+						method: req.method,
+						url: req.originalUrl,
+						statusCode: res.statusCode,
+						responseTime: durationMs,
+						httpVersion: req.httpVersion,
+						headers: req.headers,
+						query: req.query,
+					},
+				});
+			});
+
+			next();
+		});
+
```

Comment:
There's already a middleware that does the same thing as this one in order to populate endpointDurationSummary further down in this file. These should be merged into one.

Reply:
Merged in commit [0b2fcdc](https://github.com/clusterio/clusterio/pull/556/commits/0b2fcdcec471a5ba029a0f2b017ba8e35d44174b)